File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.java
class Solution {
    public int solution(int a, int b, int c) {
        int answer = 0;
        
        if(a == b && b == c && c == a) {
           return  answer = (a + b + c) * ((a * a) + (b * b) + (c * c)) * ((a * a * a) + (b * b * b) + (c * c * c));
        }else if((a == b && a != c) || (a == c && a != b) || (b == c && a != b)){
           return  answer = (a + b + c) * (a * a + b * b + c * c);
        }else{
           return answer = a + b + c;
        }
        
    }
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int a, int b, int c) {
    int answer = 0;
    
    // pow(x, n)
    if(a == b && b == c) {
        answer = (a + b + c)*(a*a + b*b + c*c)*(a*a*a + b*b*b + c*c*c);
    }else if(a == b || b == c || c == a){
        answer = (a + b + c)*(a*a + b*b + c*c);
    }else{
        answer = a + b + c;  
    }
    
    return answer;
}